src/HasSlug.php

```diff
@@ -149,8 +149,14 @@ protected function makeSlugUnique(string $slug): string
      */
     protected function otherRecordExistsWithSlug(string $slug): bool
     {
+        $key = $this->getKey() ?? '0';
+
+        if (! $this->incrementing) {
+            $key = $this->getKey();
+        }
+
         return (bool) static::where($this->slugOptions->slugField, $slug)
-            ->where($this->getKeyName(), '!=', $this->getKey() ?? '0')
+            ->where($this->getKeyName(), '!=', $key)
```

Comment:
I'm not sure what problem this solves. Could you give some extra info in the `Conversation` tab?